src/common/utils/use-waiting-message.ts

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+// Keys are the seconds to wait before showing the message
+export type WaitingMessages = {
+  [key: string]: string;
+};
+
+function messageForSecondsPassed(waitingMessages: WaitingMessages, seconds: number) {
+  return waitingMessages[seconds.toString() as keyof typeof waitingMessages];
+}
+
+export function useWaitingMessage(
```

Comment:
I like the pattern of wrapping the functionality in a hook 👍🏼 

All these values, bar the state, will be redeclared on every render. You can do this in a single memo, and keep the API the same. 

`const handlers = useMemo(() => {
  startWaitingMessage(),
  ...
});

Reply:
@kyranjamie Thanks for this, indeed my previous version was doing extra re-render. I ended up using useInterval.
It works well now (without extra rerender)